### PR TITLE
[LETS-34] Group all corruption checking related info & code into corruption_checker

### PR DESCRIPTION
**Pull request by ribram**
http://jira.cubrid.org/browse/LETS-34

Group variable and functions that deal with log page corruption into a class named corruption_checker inside log_recovery_analysis.cpp. No behavior changes.

**corruption_checker**

- Grouped variables: is_log_page_corrupted, first_corrupted_rec_lsa, null_block.
- Repalced `logpb_page_check_corruption()` with `corruption_checker::check_page_checksum()`
- Replaced `logpb_page_get_first_null_block_lsa()` with `corruption_checker::find_first_corrupted_block()`
- Replaced lines 245->291 with `corruption_checker::check_log_record()`

**log_recovery_analysis**

Variables & functions & a block of code have been replaced with corruption_checker. Also:

- Lines 184->221 have been moved to a new function called `log_rv_analysis_check_page_corruption()`. last_checked_page_id was removed and the check is applied when the record navigation advances to a new page (`record_nav_lsa.pageid != log_nav_lsa.pageid`).
- A few indentation adjustments on comments.

**log_page_buffer**

The signature of checksum related functions have been updated and simplified:

- Replaced return error code with either void or according to the function purpose, because NO_ERROR was always returned.
- Removed thread entry argument and the argument replaced by return.
- Constant log_page argument when it is not supposed to change.

**Tests**

- Simple manual tests (createdb, start database, crash & recovery)
- shell_debug

### CODE REVIEW

File: src/transaction/log_recovery_analysis.cpp [hunk 1]
```diff
@@ -2104,3 +2023,100 @@ log_recovery_notpartof_archives (THREAD_ENTRY *thread_p, int start_arv_num, cons
     }
 
 }
+
+corruption_checker::corruption_checker ()
+{
+  m_blocks_in_page_count = LOG_PAGESIZE / IO_BLOCK_SIZE;
```
Comment (cristiarg) [inline src/transaction/log_recovery_analysis.cpp:2031]: constexpr static member?

File: src/transaction/log_recovery_analysis.cpp [hunk 2]
```diff
@@ -242,84 +257,34 @@ log_recovery_analysis (THREAD_ENTRY *thread_p, INT64 *num_redo_log_records, log_
...
 		{
 		  /* Check correctness of information from log header. */
-		  LSA_COPY (&temp_log_lsa, &crt_record_lsa);
-		  temp_log_lsa.offset += sizeof (LOG_RECORD_HEADER);
-		  temp_log_lsa.offset = DB_ALIGN (temp_log_lsa.offset, DOUBLE_ALIGNMENT);
-
-		  if ((temp_log_lsa.offset > (int) LOGAREA_SIZE)
-		      || (LSA_GT (&temp_log_lsa, &first_corrupted_rec_lsa)))
+		  LOG_LSA end_of_header_lsa = crt_record_lsa;
+		  end_of_header_lsa.offset += sizeof (LOG_RECORD_HEADER);
```
Comment (cristiarg) [inline src/transaction/log_recovery_analysis.cpp:286]: offset `DB_ALIGN` ?

File: src/transaction/log_recovery_analysis.cpp [hunk 3]
```diff
@@ -346,21 +311,19 @@ log_recovery_analysis (THREAD_ENTRY *thread_p, INT64 *num_redo_log_records, log_
...
-      * page, this is not the end of the log. This situation happens when an
-      * incomplete log record is archived. Thus, its forward address is NULL.
-      * Note that we have to set lsa.pageid here since the log_lsa.pageid value
-      * can be changed (e.g., the log record is stored in two pages: an
-      * archive page, and an active page. Later, we try to modify it whenever
-      * is possible.
-      */
+       * If the next page is NULL_PAGEID and the current page is an archive page, this is not the end of the log.
+       * This situation happens when an incomplete log record is archived. Thus, its forward address is NULL.
+       * Note that we have to set lsa.pageid here since the log_lsa.pageid value can be changed (e.g., the log record
```
Comment (cristiarg) [inline src/transaction/log_recovery_analysis.cpp]: `lsa.pageid` and `log_lsa.pageid` are outdated in comment.

File: src/transaction/log_recovery_analysis.cpp (hunk 1 above)
Comment (ribram) [reply src/transaction/log_recovery_analysis.cpp:2031]: LOG_PAGESIZE is not a constexpr. Its value is determined at runtime.

File: src/transaction/log_recovery_analysis.cpp (hunk 2 above)
Comment (ribram) [reply src/transaction/log_recovery_analysis.cpp:286]: Done

File: src/transaction/log_recovery_analysis.cpp (hunk 3 above)
Comment (ribram) [reply src/transaction/log_recovery_analysis.cpp]: Done

**Review by cristiarg — APPROVED** [review]

**Pull request merged**